Content.Server/BugReports/BugReportManager.cs

```diff
@@ -0,0 +1,148 @@
+
+/// <summary>
+///     Simple manager to handle players bug reports. Will forward valid reports through <see cref="ValidPlayerBugReportReceived"/>
+/// </summary>
+public sealed class BugReportManager : IBugReportManager
+{
+    [Dependency] private readonly IServerNetManager _net = default!;
+    [Dependency] private readonly IEntityManager _entity = default!;
+    [Dependency] private readonly PlayTimeTrackingManager _playTime = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly IAdminLogManager _admin = default!;
+    [Dependency] private readonly IGameMapManager _map = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    private GameTicker _ticker = default!;
+    private bool _tickerInitialized;
+
+    public event EventHandler<ValidPlayerBugReportReceivedEvent>? ValidPlayerBugReportReceived;
+
+    /// <summary>
+    ///     List of player NetIds and the number of bug reports they have submitted this round.
+    /// </summary>
+    private readonly Dictionary<NetUserId, int> _bugReportsPerPlayerThisRound = new();
+
+    public void Initialize()
+    {
+        _net.RegisterNetMessage<BugReportMessage>(ReceivedPlayerBugReport);
+    }
+
+    private void ReceivedPlayerBugReport(BugReportMessage message)
+    {
+        // This is the only way I know to get entity systems in a manager, and it's needed for the round number and stuff.
+        // I'm using this approach to "avoid" nullability of _ticker.
+        // The code gets ugly if I have to constantly check for it being null.
+        if (!_tickerInitialized)
+        {
+            _tickerInitialized = true;
+            _ticker = _entity.System<GameTicker>();
+        }
+
+        var netId = message.MsgChannel.UserId;
+
+        if (!CheckIfValid(message, netId))
+            return;
+
+        _bugReportsPerPlayerThisRound[netId] = _bugReportsPerPlayerThisRound.GetValueOrDefault(netId) + 1;
+
+        var title = message.ReportInformation.BugReportTitle;
+        var description = message.ReportInformation.BugReportDescription;
+
+        _admin.Add(LogType.BugReport, LogImpact.High, $"{message.MsgChannel.UserName}, {netId}: submitted a bug report. Title: {title}, Description: {description}");
+
+        var bugReport = CreateBugReport(message);
+
+        ValidPlayerBugReportReceived?.Invoke(this, bugReport);
+    }
+
+    // When the round restarts, clear the dictionary.
+    public void Restart()
+    {
+        _bugReportsPerPlayerThisRound.Clear();
+    }
+
+    /// <summary>
+    ///     Checks that the given report is valid. Includes various checks on the report and the player.
+    /// </summary>
+    /// <returns>True if the report is valid, false there is an issue with the report or the user sending it.</returns>
+    private bool CheckIfValid(BugReportMessage message, NetUserId netId)
```

Comment:
You should optionally add a CVar for a minimum time between bug reports, so that people can write more than X bug reports but can only write them every Y minutes.